synapse/handlers/auth.py

```diff
@@ -205,15 +205,23 @@ def __init__(self, hs: "HomeServer"):
         # type in the list. (NB that the spec doesn't require us to do so and
         # clients which favour types that they don't understand over those that
         # they do are technically broken)
+
+        # start out by assuming PASSWORD is enabled; we will remove it later if not.
         login_types = []
-        if self._password_enabled:
+        if hs.config.password_localdb_enabled:
             login_types.append(LoginType.PASSWORD)
+
         for provider in self.password_providers:
             if hasattr(provider, "get_supported_login_types"):
                 for t in provider.get_supported_login_types().keys():
                     if t not in login_types:
                         login_types.append(t)
+
+        if not self._password_enabled:
+            login_types.remove(LoginType.PASSWORD)
```

Comment:
So this says to support password logins if:

* Password logins are enabled and one of:
    * The local DB is enabled.
    * A password provider supports password logins.

Is that correct?

Reply:
yes, basically - which matches the code used to actually check the auth submission later.